src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,246 @@
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all data from DB", e);
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars "
+                + "SET model = ?, manufacturer_id = ? "
+                + "WHERE car_id = ? AND is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update car in db. Car: " + car, e);
+        } finally {
+            deleteAllRelations(car);
+            insertDrivers(car);
+            return car;
+        }
+
+    }
+
+    private void deleteAllRelations(Car car) {
+        String query = "UPDATE cars_drivers "
+                + "SET is_deleted = true "
+                + "WHERE car_id = ? AND is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete relations with Car: " + car, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = false WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            int executeUpdate = statement.executeUpdate();
+            return executeUpdate > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car with id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT * "
+                + "FROM cars "
+                + "JOIN cars_drivers "
+                + "ON cars.id = cars_drivers.car_id "
+                + "WHERE cars_drivers.driver_id = ?;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Car car = new Car();
+                car.setId(resultSet.getLong("id"));
+                car.setModel(resultSet.getString("model"));
+                cars.add(car);

```

Comment:
think how you can refactor your code so we have only one private helper method that parses car (hint: let's fetch from db not only car, cars_drivers relations, but manufacturer as well)